sample-code/src/server.ts

```diff
@@ -51,6 +53,100 @@ app.get('/azure-openai/chat-completion', async (req, res) => {
   }
 });
 
+app.get('/azure-openai/chat-completion-stream', async (req, res) => {
+  const controller = new AbortController();
+  try {
+    const response = await chatCompletionStream(controller);
+
+    res.setHeader('Cache-Control', 'no-cache');
+    res.setHeader('Content-Type', 'text/event-stream');
+    res.setHeader('Access-Control-Allow-Origin', '*');
+    res.setHeader('Connection', 'keep-alive');
+    res.flushHeaders();
+
+    let connectionAlive = true;
+
+    res.on('close', () => {
+      controller.abort();
+      connectionAlive = false;
+      res.end();
+    });
+
+    for await (const chunk of response.stream.toContentStream()) {
+      if (!connectionAlive) {
```

Comment:
[q] what will happen, if we dont check this?
Writing empty chunks forever?
I would assume, after getting the full response, the loop should be stopped automatically?

Reply:
There will be an error if user closes the connection by e.g. refreshing the page and we still try to send something.